framework-test/src/main/java/org/checkerframework/framework/test/diagnostics/TestDiagnosticUtils.java

```diff
@@ -201,8 +211,10 @@ public static IPair<String, String> formatJavaxToolString(String original, boole
         String filename = "";
         if (noMsgText) {
             if (!retainAllLines(trimmed)) {
-                int lineSepPos = trimmed.indexOf(System.lineSeparator());
-                if (lineSepPos != -1) {
+                // When the source file doesn't use the system default line separator,
+                // System.lineSeparator() can't detect which is actually used.
+                int lineSepPos = Math.max(trimmed.indexOf("\n"), trimmed.indexOf("\r\n"));
+                if (trimmed.contains("\r\n") || trimmed.contains("\n")) {
```

Comment:
I'm still confused by this logic... The `contains` checks seem unnecessary, and the old check against `-1` would still seem sufficient: if either of them are contained, they give a >=0 indexOf, so `lineSepPos` would be > -1.
Then the `Math.max` still seems odd: Let's say `trimmed` contains `\r\n`. The index of `\n` will be larger, so `lineSepPos` will point to the `\n`.
Doesn't the `substring` then leave `trimmed` ending with the `\r`?